tests: fix unchecked return value in test_net_pkt_basics_of_rw

maxxlife opened the pull request:
Inside function test_net_pkt_basics_of_rw result of function call
net_pkt_read_be16 is not checked which might result performance
of the program and cause errors.

Coverity-CID 198003
Fixes #15776
Signed-off-by: Maksim Masalski maxxliferobot@gmail.com

jukkar commented on tests/net/net_pkt/src/main.c:
```diff
@@ -259,7 +259,11 @@ static void test_net_pkt_basics_of_rw(void)
 	net_pkt_cursor_init(pkt);
 	net_pkt_skip(pkt, 1);
-	net_pkt_read_be16(pkt, &value16);
+	if (net_pkt_read_be16(pkt, &value16) != 0) {
```
Just use proper zassert variant here instead of this check + printk.

maxxlife replied:
Maybe I'm not clear understand you, but please check my changes again.

pfalcon replied:
@maxxlife, using zassert_equal() would be even better. Just see the line below for example usage.

maxxlife committed:
tests: fix unchecked return value in test_net_pkt_basics_of_rw

Inside function test_net_pkt_basics_of_rw result of function call
net_pkt_read_be16 is not checked which might result performance
of the program and cause errors.

Coverity-CID 198003
Fixes #15776

Signed-off-by: Maksim Masalski <maxxliferobot@gmail.com>

pfalcon reviewed (approved):
LGTM, though using zassert_equal() would be better.
